TFM build fail on b_u585i_iot02a_ns

**Describe the bug**
Hi, `b_u585i_iot02a_ns` failed few builds on the latest weekly run:

`west build -p -b b_u585i_iot02a_ns  -T samples/tfm_integration/tfm_ipc/sample.tfm_ipc`

```
/usr/local/google/home/fabiobaltieri/zephyrproject/modules/tee/tf-m/trusted-firmware-m/bl2/src/shared_data.c:167:5: error: conflicting types for 'boot_save_shared_data'; have 'int(const struct image_header *, const struct flash_area *)'
  167 | int boot_save_shared_data(const struct image_header *hdr,
      |     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/google/home/fabiobaltieri/zephyrproject/modules/tee/tf-m/trusted-firmware-m/bl2/src/shared_data.c:8:
/usr/local/google/home/fabiobaltieri/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/boot_record.h:79:5: note: previous declaration of 'boot_save_shared_data' with type 'int(const struct image_header *, const struct flash_area *, const uint8_t,  const int)' {aka 'int(const struct image_header *, const struct flash_area *, const unsigned char,  const int)'}
   79 | int boot_save_shared_data(const struct image_header *hdr,
      |     ^~~~~~~~~~~~~~~~~~~~~
```

Could you look into this?

**To Reproduce**
`west build -p -b b_u585i_iot02a_ns  -T samples/tfm_integration/tfm_ipc/sample.tfm_ipc`

**Impact**
CI build fail

--- a/west.yml
+++ b/west.yml
@@ -331,7 +331,7 @@ manifest:
       groups:
         - debug
     - name: trusted-firmware-m
-      revision: 8b6146261fe2c0ad61154e20c7e338601eae2208
+      revision: b168d92c7ed3c77c94d7ce3362bdde5dbffe8424
       path: modules/tee/tf-m/trusted-firmware-m
       groups:
         - tee

